src/ui/public/share/directives/share_object_url.js

```diff
@@ -69,8 +70,14 @@ app.directive('shareObjectUrl', function (Private, Notifier) {
         });
       };
 
+      // since getUrl() is called within a watcher we cache the unhashing step
+      const unhashStatesInAbsUrl = memoize((absUrl) => {
```

Comment:
@tylersmalley How would this be affected by replacing hashes with auto-incremented integers? It might have a slightly negative performance impact because different integers could represent identical states, whereas the hashes always map to unique states.